docs/API.md

```diff
@@ -1,681 +1,1085 @@
+* `"advanced_options"`: A string or `null`. `"1"` if the `"advanced"` flag was set in the method call to `start_domain_test` that created this Test.
+* `"overall_result"`: A string. The most severe problem level logged in the test results.
+
+>
+> TODO: Describe the format of `"overall_result"`.
+>
+> TODO: What about if the *test* was created with `add_batch_job` or something else?
+>
+> TODO: What about if the *test* was created with `"advanced"` set to `false` in `start_domain_test`?
+>
+
+
+#### `"error"`
+
+>
+> TODO: List all possible error codes and describe what they mean enough for clients to know how react to them.
+>
+
+
+## API method: `add_api_user`
+
+>
+> TODO: Method description.
+>
+
+This method requires the *administrative* *privilege level*.
+
+Example request:
+```json
 {
-    "jsonrpc": "2.0",
-    "id": 4711,
-    "method": "add_api_user",
-    "params": {
-        "username": "citron",
-        "api_key": "fromage"
-    }
+  "jsonrpc": "2.0",
+  "method": "add_api_user",
+  "id": 4711,
+  "params": {
+    "username": "citron",
+    "api_key": "fromage"
+  }
 }
 ```
 
- -  params: an object containing the following parameters
-    -  username: the name of the user to add
-    -  api_key: the API key (in effect, password) for the user to add
- -   jsonrpc: « 2.0 »
- -   id: any kind of unique id allowing to match requests and responses
- -   method: the name of the called method
-
- This method implements a very basic security mechanism for the batch
- API. It will accept to create a new user only if the client is calling it from a "localhost" IP.
- 
-**Response**:
-```
+Example response:
+```json
 {
+  "id": 4711,
   "jsonrpc": "2.0",
-  "id": 4711
-  "result": 1
+  "result": 0
 }
 ```
 
- -  jsonrpc: « 2.0 »
- -  id: any kind of unique id allowing to match requests and responses
- -  result: 1 if the user was created 0 otherwise.
 
- 
-### *JSON-RPC Call*: `add_batch_job`
+#### `"params"`
 
-**Request**:
-```
+An object with the following properties:
+
+* `"username"`: A string, optional. The name of the user to add.
+* `"api_key"`: A string, optional. The API key (in effect, password) for the user to add.
+
+>
+> TODO: Are `"username"` and `"api_key"` really supposed to be optional? Because
+> they are now, is that a bug? I get `"result": 0` when I omit them. I would
+> have expected parameter validation errors.
+>
+
+
+#### `"result"`
+
+An integer.
+
+>
+> TODO: Describe the possible values of the result and what they mean.
+>
+
+
+#### `"error"`
+
+>
+> TODO: List all possible error codes and describe what they mean enough for clients to know how react to them.
+>
+
+
+## API method: `add_batch_job`
+
+>
+> TODO: Method description.
+>
+
+All the domains will be tested using identical parameters.
+
+If an identical *test* for a domain was already enqueued and hasn't been started or was enqueued less than 10 minutes earlier,
+no new *test* is enqueued for this domain.
+
```

Comment:
A "batch user" cannot start a new batch before the first batch has completed. If a second "batch user" is created, then a second batch can be started in parallel with the first batch.